File: Programmers/Level 2/멀쩡한 사각형.js
// 최대 공약수 문제;;
const greatestCommonDivisor = (a, b) => {
       return b === 0 ? a : greatestCommonDivisor(b, a % b)
    }

function solution(w, h) {
    const gcd = greatestCommonDivisor(w, h)
    return (w * h) - ((w / gcd) + (h / gcd) - 1) * gcd
}

File: Programmers/Level 2/멀쩡한 사각형.py
def greatestCommonDivisor(a, b):
    return a if b == 0 else greatestCommonDivisor(b, a % b)


def solution(w, h):
    gcd = greatestCommonDivisor(w, h)
    return (w * h) - ((w / gcd) + (h / gcd) - 1) * gcd
